hangupsbot/plugins/tldr.py

```diff
@@ -35,9 +35,16 @@ def tldrecho(bot, event, *args):
     else:
         # No path was found. Is this your first setup?
         new_tldr = 0
+    
+    if tldr_echo_options[new_tldr] is not "GLOBAL":
+        # Update the tldr_echo setting
+        bot.memory.set_by_path(['conversations', event.conv_id, 'tldr_echo'], new_tldr)
+    else:
+        # If setting is global then clear the conversation memory entry
+        conv_settings = bot.memory.get_by_path(['conversations', event.conv_id])
+        del conv_settings['tldr_echo'] # remove setting
+        bot.memory.set_by_path(['conversations', event.conv_id], conv_settings)
```

Comment:
In the current setup, removing `tldr_echo` will force the bot to set it to 0 on the next toggle, which works in this scenario, but the logic isn't ideal. Where in the code is it necessary that the tldr_echo needs to be cleared? Perhaps we can take a look at those parts and see if we can improve that without breaking the tldr_echo toggle logic.